core/WFCORE-1427_Suspend_Resume_servers_by_host_controller.adoc

```diff
@@ -0,0 +1,129 @@
+
+=== Related Issues
+
+* {issue-base-url}/WFCORE-1632[WFCORE-1632]
+* {issue-base-url}/WFCORE-1917[WFCORE-1917]
+* {issue-base-url}/WFCORE-3804[WFCORE-3804]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:yborgess@redhat.com[Yeray Borges]
+* mailto:jdenise@redhat.com[Jean-Francois Denise]
+
+=== QE Contacts
+
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Affected Projects or Components
+* WildFly Core kernel
+* WildFly Core CLI
+* HAL
+
+=== Other Interested Projects
+* N/A
+
+== Requirements
+
+=== Hard Requirements
+
+* Users should be able to suspend the servers managed by a specific host using a new host level operation called 'suspend-servers'.
+* Users should be able to resume the servers managed by a specific host using a new host level operation called 'resume-servers'.
+* Host level suspend operation should behave like any other suspend operations available for another set of servers, for example, the suspend-servers operation available at server-group level.
+* The host level operation 'suspend-servers' will have a timeout parameter named 'suspend-timeout' that controls how long the suspend operation should wait for in-flight requests to complete before returning.
+* The timeout is per server, not an aggregate overall timeout for operations that affect multiple servers.
+* Host controllers will send the suspend operation to the managed servers in parallel.
+* There are already operations which use a timeout parameter for similar purposes. The name of this parameter leads to confusion because it can be interpreted as an operation timeout. In order to have a consistency and to fix this confusion, this 'timeout' parameter will be deprecated in all relevant operations. A 'suspend-timeout' parameter will be added to be used instead of the deprecated one.
+
+=== Nice-to-Have Requirements
+* HAL support. This should be a hard requirement for a subsequent release but is a nice to have initially.
+* The use of 'suspend-timeout' param of the high level shutdown operation and adding a 'suspend-timeout' param to the low level 'shutdown' host operation in domain mode to get the host level graceful shutdown not oly via the suspend operation, but also using the shutdown operation.
+
+
+=== Non-Requirements
+* For host level 'suspend-servers' operation affecting multiple servers, treating the 'suspend-timeout' as being some overall timeout. The 'suspend-timeout' applies to each server.
+
+=== Design notes
+
+These are just examples of how the new operations looks like:
+
+- If the user wants to reload a host restarting its servers gracefully:
+
+[source]
+--
+[domain@localhost:9990 /] /host=master:suspend-servers(suspend-timeout=20)
+[domain@localhost:9990 /] /host=master:reload(restart-servers=true)
+--
+
+
+- If the user wants to shutdown gracefully a host
+
+[source]
+--
+[domain@localhost:9990 /] /host=master:suspend-servers(suspend-timeout=20)
+[domain@localhost:9990 /] /host=master:shutdown()
```

Comment:
Actually, in this case, the shutdown operation for host offers me to define `suspend-timeout` attribute. So, this can be simplified to just
```
[domain@localhost:9990 /] /host=master:shutdown(suspend-timeout=20)
```
or did I miss anything?

Reply:
You're right, the issue with this sentence is we had this possibility as a nice-to-have requirement. Since finally it was implemented, I'm going to remove it.

Reply:
Changed

Reply:
Thanks, looks good now. Sorry, I didn't realize that this was in a `nice-to-have` requirements. Thus original variant was probably also okay. Anyway, thank you for the update!